### DISCUSSION TITLE
setup preview apps with external Heroku API backend

### DISCUSSION THREAD

**Discussion by lifeiscontent** [vercel/community#130] [Show & Tell]
Hi, I have an app hosted on Vercel, which is consuming an API on Heroku.

Both Vercel and Heroku have a concept of preview apps, I'm wondering if its possible to pair them together.

the code exists here:

https://github.com/lifeiscontent/realworld

as you can see, there are 2 separate folders, api is deployed to Heroku while web is deployed to Vercel.

I'm wonder if there's a way I can set it up so when I create a PR, Heroku is deployed, once it's finished, it tells Vercel "Hey, here's the URL for the API preview app" and then Vercel deploys the web folder with the right API_HOST env var.

is this possible? if so, how would I go about doing something like this?

if not, suggestions on how to simplify this process are welcomed.

Thanks!

**Comment by Pieparker** [comment]
I'm not too au fait with Heroku however you can certainly trigger branch deploys using [deploy hooks](https://vercel.com/docs/concepts/deployments/overview#deploy-hooks). You have a greater degree of flexibility when triggering builds via the [CLI ](https://vercel.com/docs/concepts/deployments/overview#vercel-cli) or [API](https://vercel.com/docs/concepts/deployments/overview#vercel-api), though!
